documentation/fancylogger/Opt-In-Mechanism.md

```diff
@@ -0,0 +1,21 @@
+# When should we use FancyLogger
+
+The FancyLogger presents the user with the build's most relevant information at the time, automatically hiding all the information that is no longer relevant (as to prevent huge log outputs). However, many users might find this not very useful (or even counterproductive) such as those using a terminal without proper ANSI support or when redirecting the output to a file. For that reason, the users must be able to turn this feature on/off at will.
+
+# Proposal
+Using the `/fancylogger` or `/flg` command line switches, users are able to opt-in and use the FancyLogger, EXCEPT when:
+- The terminal does not support ANSI codes or color
+    - [ ] Detect if terminal does not support ANSI codes or color
+ - Output is redirected to a file or pipe
+    - [x] Detect if terminal output is redirected
+
+For early development stages, an environment variable `$MSBUILDFANCYLOGGER` should be enabled to prevent accidental access to an unfinished feature. 
+
+In cases where the FancyLogger should not be enabled, the default ConsoleLogger should be used instead.
```

Comment:
We should probably log a deferred build message that we're ignoring the FancyLogger opt-in because {condition}.

Reply:
Agreed, what about something like:
`Attached FancyLogger because the /fancylogger switch was used`
or
`Attached FancyLogger because the environment variable $MSBUILDFANCYLOGGER was used`
?

Reply:
And also 
`Attached ConsoleLogger because output is not compatible (ie, it does not support ANSI sequences or is a file)`

Reply:
I am thinking mostly about the negative cases here, where we were told to turn FancyLogger on but didn't. The log should already have all the command line params and all environment variables starting with MSBUILD so I don't know if we need to log in that case.